lib/pubid/iso/harmonized_stage_code.rb

```diff
@@ -74,32 +83,63 @@ class HarmonizedStageCode
       proceed: "99",
     }.freeze
 
-    def initialize(stage, substage = "00")
-      # when stage is stage name
-      if STAGES_NAMES.key?(stage)
-        @stage = STAGES_NAMES[stage]
-        @substage = SUBSTAGES_NAMES[substage]
+    # @param stage_or_code [String,Array<String>] stage number or whole harmonized code with substage
+    #   or list or stages for fuzzy stages eg. "10", 10, "20.20", ["10.20", "20.20"]
+    # @param substage [Integer, String] eg. "00", 0
+    def initialize(stage_or_code, substage = "00")
+      @stages = if stage_or_code.is_a?(Array)
+                  stage_or_code
+                elsif stage_or_code.is_a?(String) && DESCRIPTIONS.key?(stage_or_code)
+                  [stage_or_code]
+                # when stage is stage name
+                elsif STAGES_NAMES.key?(stage_or_code)
+                  ["#{STAGES_NAMES[stage_or_code]}.#{SUBSTAGES_NAMES[substage]}"]
+                else
+                  # stage is number
+                  ["#{stage_or_code}.#{substage}"]
+                end
+      validate_stages
+    end
+
+    def validate_stages
+      @stages.each do |stage|
+        # raise an error when stage is wrong
+        raise Errors::HarmonizedStageCodeInvalidError unless DESCRIPTIONS.key?(stage)
+      end
+    end
+
+    def to_s
+      if fuzzy?
+        return "draft" if @stages.all? { |s| DRAFT_STAGES.include?(s) }
+
+        return "canceled" if @stages.all? { |s| CANCELED_STAGES.include?(s) }
+
+        return "published" if @stages.all? { |s| PUBLISHED_STAGES.include?(s) }
+
+        raise Errors::HarmonizedStageRenderingError, "cannot render fuzzy stages"
       else
-        # stage is number
-        @stage, @substage = stage, substage
-        validate_stage(stage, substage)
+        @stages.first
       end
     end
 
-    def validate_stage(stage, substage)
-      # raise an error if stage is not number
-      raise Errors::HarmonizedStageCodeInvalidError if Integer(stage).nil?
+    def ==(other)
```

Comment:
You may also like to add the `<>` method to allow checking if a "00.00" is within the "00" stage, etc.

Reply:
> You may also like to add the `<>` method to allow checking if a "00.00" is within the "00" stage, etc.

Now there is no way to compare it without substage. How do you imagine it?

Reply:
I was thinking of sorting by stages and then substages within stages. But clearly it's not important here and shouldn't block the merge.